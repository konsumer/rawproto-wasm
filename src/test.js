const { parse_raw } = require('../pkg/rawproto.js')

function getString (binary, path) {
  const entries = parse_raw(binary, path, { no_fixed64: true, no_fixed32: true })
  return entries.map(entry => {
    return String.fromCharCode(...entry.value.Bytes)
  })
}

describe('rawproto', () => {
  test('basic parsing', () => {
    const opt = { no_fixed64: true, no_fixed32: false }
    expect(parse_raw(Buffer.from('\x12\x07Unknown'), '', opt)).toMatchSnapshot()
    expect(parse_raw(Buffer.from('\x12\x07Unknown\x12\x07Unknown'), '', opt)).toMatchSnapshot()
  })

  test('string parsing', () => {
    expect(getString(Buffer.from('\x12\x07Unknown\x12\x07Unknown'), '.2')).toMatchSnapshot()
  })
})
