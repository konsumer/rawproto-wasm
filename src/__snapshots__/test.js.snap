// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rawproto basic parsing 1`] = `
Array [
  Object {
    "path": Array [
      2,
    ],
    "value": Object {
      "Bytes": Array [
        85,
        110,
        107,
        110,
        111,
        119,
        110,
      ],
    },
  },
]
`;

exports[`rawproto basic parsing 2`] = `
Array [
  Object {
    "path": Array [
      2,
    ],
    "value": Object {
      "Bytes": Array [
        85,
        110,
        107,
        110,
        111,
        119,
        110,
      ],
    },
  },
  Object {
    "path": Array [
      2,
    ],
    "value": Object {
      "Bytes": Array [
        85,
        110,
        107,
        110,
        111,
        119,
        110,
      ],
    },
  },
]
`;

exports[`rawproto string parsing 1`] = `
Array [
  "Unknown",
  "Unknown",
]
`;
